jet_images_ml.py

```diff
@@ -116,170 +72,202 @@ def flatten_layer(layer):
+    for i in range(starting_iteration,
+                   starting_iteration + num_iterations):
+
         # Get a batch of training examples.
         # x_batch now holds a batch of images and
         # y_true_batch are the true labels for those images.
 
-        x_batch, y_true_batch, _, cls_batch = data.train.next_batch(train_batch_size)
-        x_valid_batch, y_valid_batch, _, valid_cls_batch = data.valid.next_batch(train_batch_size)
-       
-        # Convert shape from [num examples, rows, columns, depth]
-        # to [num examples, flattened image shape]
-        
-        x_batch = x_batch.reshape(train_batch_size, img_size_flat)
-        
-        x_valid_batch = x_valid_batch.reshape(train_batch_size, img_size_flat)
+        x_batch, y_true_batch, _, cls_batch = data.train.next_batch(batch_size)
+        x_valid_batch, y_valid_batch, _, valid_cls_batch = data.valid.next_batch(batch_size)
+
+        if not convolutional:
+            # Convert shape from [num examples, rows, columns, depth]
+            # to [num examples, flattened image shape]
+            x_batch = x_batch.reshape(batch_size, img_size_flat)
+            x_valid_batch = x_valid_batch.reshape(batch_size, img_size_flat)
+
         # Put the batch into a dict with the proper names
         # for placeholder variables in the TensorFlow graph.
-        
-        
-        feed_dict_train = {x: x_batch,y_true: y_true_batch}
-        feed_dict_validate = {x: x_valid_batch,y_true: y_valid_batch}
+
+        feed_dict_train = {x: x_batch, y_true: y_true_batch}
+        feed_dict_validate = {x: x_valid_batch, y_true: y_valid_batch}
 
         # Run the optimizer using this batch of training data.
         # TensorFlow assigns the variables in feed_dict_train
         # to the placeholder variables and then runs the optimizer.
-        
-                    
-        session.run(optimizer, feed_dict=feed_dict_train)
-        #saver = tf.train.Saver()
-        #saver.save(session, 'my_test_model') 
-        train_acc = session.run(accuracy, feed_dict=feed_dict_train)
-        validate_acc = session.run(accuracy, feed_dict=feed_dict_validate)
-        
-        av_train_acc = av_train_acc+ train_acc 
-        
-        
-        av_validate_acc =av_validate_acc+validate_acc 
-        
-        # Print status at end of each epoch (defined as full pass through training dataset).
-        val_loss  = session.run(cost, feed_dict=feed_dict_validate)
-        av_val_loss =av_val_loss+val_loss
-        
-        #loss,val_loss= session.run([merged_summary,cost], feed_dict=feed_dict_validate)
-        
- 
-        
-        #print (av_train_acc,av_validate_acc,av_val_loss)
-        if i % int(data.train.num_examples/batch_size) == 0: 
-            
-            a=session.run(merged_summary,feed_dict_train)
-            writer.add_summary(a,i)
 
+        # training step
+        _, train_acc, train_loss = session.run([optimizer, accuracy, cost], feed_dict=feed_dict_train)
```

Comment:
To extract the features, I added the following

_, train_acc, train_loss,classifier_features = session.run([optimizer, accuracy, cost,sigmoid], feed_dict=feed_dict_train)
classifier.append(classifier_features)
and at the end of the loop (above 270),

features_class=np.concatenate(classifier,axis=0)
features_classifier=pd.DataFrame(features_class)
features_classifier.to_csv("Features_Pythia_500.csv",index=False)

where classifier=[] defined before the for loop

Am I doing the right thing?